### ISSUE TITLE
Problem with complex fix using `paste` + `replace_all`

### ISSUE THREAD

**Issue by TobiasNx** [metafacture/metafacture-fix#278]
https://github.com/hbz/lobid-resources/pull/1584

I have a complex fix that uses paste and replace_all besides others.
[while the paste alone does not create the wrong transformation ](https://github.com/hbz/lobid-resources/pull/1584/commits/7c2f749976501b8fc505ee2190449ac650469e5b)
[
Adding an replace_all function transforming the results of the pase seem to create an wrong output:](https://github.com/hbz/lobid-resources/pull/1584/commits/72bfc00b530b4f07afd8b7ccd4eb905c9053a875)

```
    "altLabel" : [ "Ev.-Luth. Landeskirche Sachsens ( )", "Evang.-Luth. Landeskirche Sachsens ( )", "EVLKS ( )", "Ev.-Luth. Landeskirche Sachsens ( )", "Evang.-Luth. Landeskirche Sachsens ( )", "EVLKS ( )" ],
    "$i" : {
      "altLabel" : [ {
        "1" : "Ev.-Luth. Landeskirche Sachsens",
        "2" : "Evang.-Luth. Landeskirche Sachsens",
        "3" : "EVLKS",
        "4" : "Ev.-Luth. Landeskirche Sachsens",
        "5" : "Evang.-Luth. Landeskirche Sachsens",
        "6" : "EVLKS"
      } ]
      ```

I use the combination with paste and replace_all at another place in the fix but there do not have the problem:
https://github.com/hbz/lobid-resources/pull/1584/commits/733286d2e1f6841f795650c57c18c166c453ce06

**Comment by blackwinter** [comment]
`replace_all()` seems to have issues with programmatically added values in arrays (not limited to `paste()`).

**Comment by blackwinter** [comment]
Here are a bunch of test cases: bb96de7

@fsteeg: Can you look into fixing the path logic?

@TobiasNx: Do you want to try and supply an integration test for your use case?

**Comment by TobiasNx** [comment]
Is this also related to this problem: https://github.com/hbz/lobid-resources/pull/1585

**Comment by blackwinter** [comment]
> Is this also related to this problem: [hbz/lobid-resources#1585](https://github.com/hbz/lobid-resources/pull/1585)?

Yes.

> Maybe it is related to this: #250

Doesn't look like it, but can't say for sure.

**Comment by fsteeg** [comment]
> @fsteeg: Can you look into fixing the path logic?

I'll take a look, thanks for the unit tests!

**Comment by fsteeg** [comment]
Pushed some tweaks in [87e30a7](https://github.com/metafacture/metafacture-fix/commit/87e30a742c7cf24bb13d946ada59c33cb0bb932f) and [416ba75](https://github.com/metafacture/metafacture-fix/commit/416ba75bb57b8589492bbd0e865bb511a5acb261), which seem to fix the use cases from the unit tests.

**Comment by TobiasNx** [comment]
I added my case here as integration test: https://github.com/metafacture/metafacture-fix/pull/280

**Comment by TobiasNx** [comment]
Tested my integration test it works in the new branch.
